src/components/organisms/AssetActions/Edit/EditComputeDataset.tsx

```diff
@@ -0,0 +1,189 @@
+  content.form.data.find(
+    (data: { name: string }) => data.name === 'publisherTrustedAlgorithms'
+  ).options = algorithmOptions
+
+  const { ocean } = useOcean()
+  const { accountId } = useWeb3()
+  const { ddo, refreshDdo } = useAsset()
+  const [success, setSuccess] = useState<string>()
+  const [error, setError] = useState<string>()
+
+  const hasFeedback = error || success
+
+  async function handleSubmit(
+    values: ServiceComputePrivacy,
+    resetForm: () => void
+  ) {
+    try {
+      let trustedAlgorithms: PublisherTrustedAlgorithm[] = []
+
+      trustedAlgorithms = await createTrustedAlgorithmList(
+        values.publisherTrustedAlgorithms,
+        ocean
+      )
+
+      const privacy: ServiceComputePrivacy = {
+        allowRawAlgorithm: false,
+        allowNetworkAccess: values.allowNetworkAccess,
+        publisherTrustedAlgorithms: trustedAlgorithms
+      }
+
+      const ddoEditedComputePrivacy = await ocean.compute.editComputePrivacy(
+        ddo,
+        1,
+        privacy
+      )
+
+      if (!ddoEditedComputePrivacy) {
+        setError(content.form.error)
+        Logger.error(content.form.error)
+        return
+      }
+
+      const storedddo = await ocean.assets.updateMetadata(
+        ddoEditedComputePrivacy,
+        accountId
+      )
+      if (!storedddo) {
+        setError(content.form.error)
+        Logger.error(content.form.error)
+        return
+      } else {
+        // Edit succeeded
+        setSuccess(content.form.success)
+        resetForm()
+      }
+    } catch (error) {
+      Logger.error(error.message)
+      setError(error.message)
+    }
+  }
+
+  return (
+    <Formik
+      initialValues={getInitialValues(
+        ddo.findServiceByType('compute').attributes.main.privacy
+      )}
+      validationSchema={validationSchema}
+      onSubmit={async (values, { resetForm }) => {
+        // move user's focus to top of screen
+        window.scrollTo({ top: 0, left: 0, behavior: 'smooth' })
+        // kick off editing
+        await handleSubmit(values, resetForm)
+      }}
+    >
+      {({ isSubmitting, initialValues }) =>
+        isSubmitting || hasFeedback ? (
+          <MetadataFeedback
+            title="Updating Data Set"
+            error={error}
+            success={success}
+            setError={setError}
+            successAction={{
+              name: content.form.successAction,
+              onClick: async () => {
+                await refreshDdo()
+                setShowEdit(false)
+              }
+            }}
+          />
+        ) : (
+          <>
+            <p className={styles.description}>{content.description}</p>
+            <article className={styles.grid}>
+              <>
+                <FormEditComputeDataset
+                  title={content.form.title}
+                  data={content.form.data}
+                  setShowEdit={setShowEdit}
+                  values={initialValues}
```

Comment:
this is the trick I think, we are modifying in the `FormEditComputeDataset` subcomponent those `values` but these are not the `values` we are looking for. The actual form `values` should be on here https://github.com/oceanprotocol/market/pull/417/files#diff-12dc7146bf292ba49087391d56cba42ed37a4b4339410e318535a430235225ffR155

We then pass those as `values`. And we pass `initialValues` as a new prop `initialValues`